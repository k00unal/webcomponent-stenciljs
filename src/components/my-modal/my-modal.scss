my-modal {
  background-color: #fff;
  border: 1px solid #000;
  box-shadow: 1px 1px 1px #000;
  display: none;
  font-family: Arial, Helvetica, sans-serif;
  left: calc(50% - 175px);
  position: fixed;
  text-align: center;
  top: 30%;
  width: 350px;
  z-index: 1000;
}

.wc-modal {
  &__wrap {
    position: relative;
  }

  &__cls {
    position: absolute;
    right: 0;
    top: -21px;
  }
}
