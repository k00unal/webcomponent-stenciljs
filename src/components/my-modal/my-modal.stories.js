import { storiesOf } from "@storybook/html";

storiesOf("my-modal", module).add(
  "Default",
  () => `<button id="modal-btn">Modal button</button>
<my-modal style="display:block"
  class="wc-modal"
  name="Simple modal"
  bodytxt="This is a simple Modal"
></my-modal>`,
  {}
);
