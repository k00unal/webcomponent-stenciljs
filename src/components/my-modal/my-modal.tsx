import { Component, h, Method, Element, Prop } from "@stencil/core";

@Component({
  tag: "my-modal",
  styleUrl: "my-modal.scss"
})
export class MyModal {
  @Element() modalEl: HTMLElement;

  @Prop() name: string;
  @Prop() bodytxt: string;

  @Method()
  async open() {
    this.modalEl.style.display = "block";
  }

  hideButtonHandler() {
    this.modalEl.style.display = "none";
  }

  render() {
    return (
      <div class="wc-modal__wrap">
        <button
          class="wc-modal__cls"
          onClick={this.hideButtonHandler.bind(this)}
        >
          X
        </button>
        <h1>{this.name}</h1>
        <p>{this.bodytxt}</p>
      </div>
    );
  }
}
