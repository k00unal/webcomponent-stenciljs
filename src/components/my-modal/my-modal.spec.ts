import { newE2EPage } from "@stencil/core/testing";
const puppeteer = require("puppeteer");
import { MyModal } from "./my-modal";

describe("<my-modal>", () => {
  it("should build", () => {
    expect(new MyModal()).toBeTruthy();
  });
});
describe("rendering", () => {
  it("should have h1 with name", async () => {
    const page = await newE2EPage();
    await page.setContent(`<my-modal></my-modal>`);
    const el = await page.find("my-modal h1");
    expect(el).not.toBeNull();
  });

  // Snapshot testing
  it("should populate the name attribute", async () => {
    const page = await newE2EPage({
      html: `
    <my-modal name="Simple modal"></my-modal>
    `
    });

    const BulletLinkComponent = await page.find("my-modal");
    await page.waitForChanges();
    expect(BulletLinkComponent.innerHTML).toMatchSnapshot();
  });
});

//create global variables to be used in the beforeAll function
let browser;
let page;

beforeAll(async () => {
  // launch browser
  browser = await puppeteer.launch({
    headless: false, // headless mode set to false so browser opens up with visual feedback
    slowMo: 80 // how slow actions should be
  });
  // creates a new page in the opened browser
  page = await browser.newPage();
});

describe("E2E browser modal component check", () => {
  it("Should check for h1 tag content", async () => {
    await page.goto("http://localhost:3333/");
    await page.click("#modal-btn");
    const finalText = await page.$eval("h1", el => el.textContent);
    expect(finalText).toBe("Simple modal");
  }, 10000);
  it("Should check for h1 tag content", async () => {
    await page.goto("http://localhost:3333/");
    await page.click("#modal-btn");
    const finalText = await page.$eval("p", el => el.textContent);
    expect(finalText).toBe("Simple modal");
  }, 10000);
});

// This function occurs after the result of each tests, it closes the browser
afterAll(() => {
  browser.close();
});

// test("Should check for h1 tag inside the modal component", async () => {
//   const browser = await puppeteer.launch({
//     headless: false,
//     slowMo: 80
//     // args: ["--window-size=1920,1080"]
//   });
//   const page = await browser.newPage();
//   await page.goto("http://localhost:3333/");
//   await page.click("#modal-btn");
//   const finalText = await page.$eval("h1", el => el.textContent);
//   expect(finalText).toBe("Simple modal");
// });
