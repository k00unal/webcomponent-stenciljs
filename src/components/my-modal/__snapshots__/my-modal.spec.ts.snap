// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should populate the name attribute 1`] = `"<div class=\\"wc-modal__wrap\\"><button class=\\"wc-modal__cls\\">X</button><h1>Simple modal</h1><p></p></div>"`;
